compiler/main/driver.cpp

```diff
@@ -1157,6 +1173,11 @@ static void setupChplGlobals(const char* argv0) {
     envMap["CHPL_HOME"] = CHPL_HOME;
   }
 
+  // tell printchplenv that we're doing an LLVM build
+  if (llvmCodegen) {
+    envMap["CHPL_LLVM_CODEGEN"] = "llvm";
```

Comment:
Couldn't think just set `envMap["CHPL_TARGET_COMPILER"] = "clang-included"` and get rid of most of the chplenv changes?

Reply:
I got rid of many of the chplenv changes but it's important for e.g. detecting target architecture that the "original compiler" (i.e. what we would use with the C backend) is available as well as clang-included. In order to do that, we used to effectively run printchplenv twice, but by making the small change I made to it we can run it just once now.

Reply:
If we're doing an `--llvm` compile, why is the "original" target arch needed?

Reply:
Because we detect Cray stuff with PrgEnv compilers, e.g. https://github.com/chapel-lang/chapel/blob/master/util/chplenv/chpl_arch.py#L286

Reply:
I'm not following. If llvm is being used, why should a user be required to have a cray-prgenv loaded? Shouldn't we update chpl_arch to check for CRAY_CPU_TARGET if a prgenv is being used or if we're on a cray and clang-included is being used?

Also can we assume that clang/llvm accepts the cray arch names?